backend/core/src/listings/listings.service.ts

```diff
@@ -27,6 +27,8 @@ export class ListingsService {
       .leftJoinAndSelect("units.amiChart", "amiChart")
       .leftJoinAndSelect("listings.jurisdiction", "jurisdiction")
       .leftJoinAndSelect("listings.reservedCommunityType", "reservedCommunityType")
+      .leftJoin("listings.applications", "applications")
```

Comment:
What is the performance overhead of this join? Because it's probably joining ~1.5k rows for some listings and might result in the memory problem we faced some time ago. Please take a look how `totalFlagged` is implemented here https://github.com/bloom-housing/bloom/blob/master/backend/core/src/application-flagged-sets/application-flagged-sets.service.ts#L44 and maybe consider going with COUNT query over applications with specified listing_id foreign key.

Reply:
Ah, yes, I like this approach better. I'll update.

Reply:
I removed this and added logic to listing service to fetch the application counts and map them to the listings. I also pulled in the origin header, so that we only do this on partners, so there is a new env variable for partners URL.